notifications/models/govuk-notify-model.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable consistent-return */
+const config = require('../../config');
+const logger = require('../../lib/logger')({ env: config.env });
+const {
+  caseworkerEmail,
+  successTemplateId,
+  failureTemplateId
+} = config.notifications;
+
+module.exports = class NotifyModel {
+  async sendCaseworkerNotification(client, jobReport) {
+    const {
+      success,
+      recordsCount,
+      errorMessage,
+      fileUploadTime,
+      jobEndedTime,
+      invalidRecords
+    } = jobReport;
+
+    const emailProps = {
+      csv_uploaded_datetime: this.getDateAndTimeString(fileUploadTime),
+      cepr_update_datetime: this.getDateAndTimeString(jobEndedTime),
+      has_invalid_records: (invalidRecords.length ? 'yes' : 'no'),
+      link_to_file: ''
+    };
+    let templateId;
+    if (success) {
+      emailProps.records_count = recordsCount;
+      templateId = successTemplateId;
+    } else {
+      emailProps.failure_message = errorMessage;
+      templateId = failureTemplateId;
+    }
+
+    if (invalidRecords.length) {
+      const invalidRecordsCsv = Buffer.from(this.writeInvalidRecordsToCsv(invalidRecords), 'utf8');
+      const fileNameDate = this.getDateAndTimeString(fileUploadTime).replace(/[\/\s:]/g, '-');
+      emailProps.link_to_file = client.prepareUpload(invalidRecordsCsv, {
+        filename: `invalid-cepr-records-uploaded-${fileNameDate}.csv`
+      });
+    }
+
+    try {
+      return await client.sendEmail(templateId, caseworkerEmail, {
+        personalisation: emailProps
+      });
+    } catch (error) {
+      logger.log('error', 'Error sending notifications:', error);
+    }
+  }
+
+  createCsvHeadersFromFirstRecord(invalidRecord) {
+    const { record } = invalidRecord;
+    return `${Object.keys(record).join(',')},Reasons for invalidity`;
```

Comment:
```suggestion
    return `${Object.keys(record).join(',')},Reasons for invalid record`;
```
invalidity is just a bit wordy and might not be specific enough. Also should you mention this is related to the CSV?

Reply:
it's more about readability and cognitive load for a developer.  It's fine from a grammar perspective.  It's really no biggie so feel free to keep it 🙂